src/App.js

```diff
@@ -52,8 +52,10 @@ export const initialized = (async () => {
   initEvents()
 
   return localStateLoaded
+}
 
-})()
+// Note: Manually initialize during test environment
+if (process.env.NODE_ENV !== 'test') initialize()
```

Comment:
What if we called `initialize` in onComponentMount of the `App` component? Then it would only load when the `App` itself loaded, and we could avoid test detection.

Reply:
`onComponentMount` will run after we do `wrapper.update`. It will cause `initialize` to run twice. The problem is that initialization logic runs as soon as we import component from `src/App.js`. We should separate all the initialization logic to a separate file and not export `App` component from the same file. We can call initialization logic in `index.js` and import and render `App` directly from the components folder. This way we can separately call and wait for initialization logic to complete in test and then mount `App` component.

Reply:
Yes! I like that plan. Would you be willing to do that?

If I understand correctly, `initialize` would go into something like `/src/util/initialize.js`, `/src/index.js` would invoke `initialize`, and `/src/App.js` would be removed completely.

Reply:
> If I understand correctly, `initialize` would go into something like `/src/util/initialize.js`, `/src/index.js` would invoke `initialize`, and `/src/App.js` would be removed completely.

Yes.

Reply:
> Yes! I like that plan. Would you be willing to do that?

Sure. I can do that.